planner/core/partition_pruner_test.go

```diff
@@ -86,6 +86,83 @@ func (s *testPartitionPruneSuit) TestHashPartitionPruner(c *C) {
 	}
 }
 
+func (s *testPartitionPruneSuit) TestRangeColumnPartitionPruningForIn(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("drop database if exists test_range_col_in")
+	tk.MustExec("create database test_range_col_in")
+	tk.MustExec("use test_range_col_in")
+	tk.MustExec(`set @@session.tidb_enable_list_partition = 1`)
+	tk.MustExec("set @@session.tidb_partition_prune_mode='static'")
+
+	// case in issue-26739
+	tk.MustExec(`CREATE TABLE t1 (
+		id bigint(20)  NOT NULL AUTO_INCREMENT,
+		dt date,
+		PRIMARY KEY (id,dt))
+		PARTITION BY RANGE COLUMNS(dt) (
```

Comment:
Please add more tests about `PARTITION BY RANGE COLUMNS(a,d,c)`(partition by more columns). Rest LGTM

Reply:
Multiple-column range partitioning is not supported now~:
```
mysql> create table tx (
    -> a int, 
    -> b int,
    -> c int
    -> ) partition by range columns (a, b, c) (
    -> partition p0 values less than (10, 10, 10),
    -> partition p1 values less than (20, 20, 20)
    -> );
Query OK, 0 rows affected, 1 warning (0.01 sec)

mysql> show warnings;
+---------+------+---------------------------------------------------------+
| Level   | Code | Message                                                 |
+---------+------+---------------------------------------------------------+
| Warning | 8200 | Unsupported partition type RANGE, treat as normal table |
+---------+------+---------------------------------------------------------+
1 row in set (0.00 sec)

mysql> show create table tx;
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                       |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tx    | CREATE TABLE `tx` (
  `a` int(11) DEFAULT NULL,
  `b` int(11) DEFAULT NULL,
  `c` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```